coredns/endpointslice/map.go

```diff
@@ -86,9 +95,11 @@ func (m *Map) GetDNSRecords(hostname, cluster, namespace, name string, checkClus
 	}
 }
 
-func NewMap() *Map {
+func NewMap(localClusterID string, kubeConfig *rest.Config) *Map {
```

Comment:
It's better to pass in a `kubernetes.Interface` - we avoid passing `rest.Config` b/c there's no mock/fake implementation for unit tests.

```suggestion
func NewMap(localClusterID string, kubeClient kubernetes.Interface) *Map {
```